File: runtime/src/main/java/com/siimkinks/sqlitemagic/ExprN.java
package com.siimkinks.sqlitemagic;

import java.util.ArrayList;
import java.util.Collections;

import androidx.annotation.NonNull;

final class ExprN extends Expr {
  @NonNull
  private final String[] evalArgs;

  ExprN(@NonNull Column<?, ?, ?, ?, ?> column, @NonNull String op, @NonNull String[] evalArgs) {
    super(column, op);
    this.evalArgs = evalArgs;
  }

  @Override
  void addArgs(@NonNull ArrayList<String> args) {
    Collections.addAll(args, evalArgs);
  }
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/ExprN.kt
package com.siimkinks.sqlitemagic

internal class ExprN(
  column: Column<*, *, *, *, *>,
  op: String,
  private val evalArgs: Array<String>
) : Expr(
  column = column,
  op = op
) {
  override fun addArgs(args: ArrayList<String?>) {
    evalArgs.forEach(args::add)
  }
}
